components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -73,6 +90,115 @@ public Response getGroup(@PathParam(SCIMConstants.CommonSchemaConstants.ID) Stri
+                                          @HeaderParam(SCIMProviderConstants.ACCEPT_HEADER) String outputFormat,
+                                          @QueryParam(SCIMProviderConstants.ATTRIBUTES) String attribute,
+                                          @QueryParam(SCIMProviderConstants.EXCLUDE_ATTRIBUTES) String excludedAttributes,
+                                          String resourceString) {
+
+        String userName = SupportUtils.getAuthenticatedUsername();
+        try {
+            // content-type header is compulsory in post request.
+            if (inputFormat == null) {
+                throw new FormatNotSupportedException("Content type: " + SCIMProviderConstants.CONTENT_TYPE +
+                        "  not present in the request header");
+            }
+
+            if (!isValidInputFormat(inputFormat)) {
+                throw new FormatNotSupportedException("Input format: " + inputFormat + " is not supported.");
+            }
+
+            if (!isValidOutputFormat(outputFormat)) {
+                throw new FormatNotSupportedException("Output format: " + outputFormat + " is not supported.");
+            }
+        } catch (FormatNotSupportedException e) {
+            return handleFormatNotSupportedException(e);
+        }
+
+        Map<String, String> requestAttributes = new HashMap<>();
+        requestAttributes.put(SCIMProviderConstants.ID, id);
+        requestAttributes.put(SCIMProviderConstants.AUTHORIZATION, userName);
+        requestAttributes.put(SCIMProviderConstants.HTTP_VERB, PUT.class.getSimpleName());
+        requestAttributes.put(SCIMProviderConstants.RESOURCE_STRING, resourceString);
+        requestAttributes.put(SCIMProviderConstants.ATTRIBUTES, PERMISSIONS);
+        requestAttributes.put(SCIMProviderConstants.EXCLUDE_ATTRIBUTES, excludedAttributes);
+        return processRequest(requestAttributes);
+
+    }
+
+    @PATCH
+    @Path("{id}/permissions")
+    @Produces({MediaType.APPLICATION_JSON, SCIMProviderConstants.APPLICATION_SCIM_JSON})
+    public Response patchPermissionForGroup(@PathParam(SCIMConstants.CommonSchemaConstants.ID) String id,
+                                          @HeaderParam(SCIMProviderConstants.AUTHORIZATION) String authorizationHeader,
+                                          @HeaderParam(SCIMProviderConstants.CONTENT_TYPE) String inputFormat,
+                                          @HeaderParam(SCIMProviderConstants.ACCEPT_HEADER) String outputFormat,
+                                          @QueryParam(SCIMProviderConstants.ATTRIBUTES) String attribute,
+                                          @QueryParam(SCIMProviderConstants.EXCLUDE_ATTRIBUTES) String excludedAttributes,
+                                          String resourceString) {
+
+        String userName = SupportUtils.getAuthenticatedUsername();
+        try {
+            // content-type header is compulsory in post request.
+            if (inputFormat == null) {
```

Comment:
Throwing and catching in the same try-catch is a good practice. handleFormatNotSupportedException directly instead of throwing and catching.

if (!isValidOutputFormat(outputFormat)) {
return handleFormatNotSupportedException(e);
}